arangod/Graph/AttributeWeightShortestPathFinder.cpp

```diff
@@ -314,16 +346,43 @@ bool AttributeWeightShortestPathFinder::shortestPath(
 }
 
 void AttributeWeightShortestPathFinder::inserter(
-    std::vector<std::unique_ptr<Step>>& result, std::string_view s,
-    std::string_view t, double currentWeight, EdgeDocumentToken&& edge) {
+    std::vector<std::unique_ptr<Step>>& result, std::string_view prevVertexId,
+    std::string_view nextVertexId, double currentWeight,
+    EdgeDocumentToken&& edge) {
```

Comment:
I do not think this is correct.
As far as I see this will have a global _smartValue.
And all paths ever visited need to follow this one _smartValue.
However if I would start a ShortestPath search from a Satellite collection, I can actually have multiple valid _smartValues.

Source(Sat) -> (v: A) -> (v2: A) -> Target(Sat)
Source (Sat) -> (v3: B) -> (v4: B) -> Target(Sat)

Would both be valid shortest paths between two Satellite vertices.
Also I think this algorithm will now bind itself to one SmartValue only very early, and does not know if the shortest path is actually over this one value.

If not there we need to add some tests Sat -> (Shards*) -> Sat on shortest path algorithms.

Reply:
I remember having made such tests...